plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
On Tue, May 17, 2016 at 04:26:26PM -0700, Michael Grosser wrote:
> >      def deployment_labels
> > -      # Deployment labels should not include the ids of the release, role or deploy groups
> > -      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
> > +      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
> 
> yes, having base labels for matching and then adding to them sounds clean.
> 
> Do you need labels other than project and role (what can be statically added to the yml files) in the deployment or selector ? 

No, I don't :-)

> 
> If yes, then we can stop adding labels and avoid all the mess (maybe still have additional labels for pods, but no longer override labels on anything else)

There is one issue that we may want to ignore, but at least decide to ignore and
not without noticing.

The selector is optional, see [1] (it says required: false). If no selector is
specified in the yaml, then the ones from the template are used IIRC.

So, we may want to add a selector if there is none, specifying the role and
project labels (maybe deploy_group too).

That way we are really nice with everyone, I guess.

Or we can force to use a selector and be done with it (but validate it, because
no selector is valid for k8s). The ugly thing about this is that not all yamls valid
for k8s are valid for samson.

[1]: http://kubernetes.io/docs/api-reference/extensions/v1beta1/definitions/#_v1beta1_deploymentspec